### DISCUSSION TITLE
503 Response object too large when trying to pull

### DISCUSSION THREAD

**Discussion by Hartorn** [community/community#26568] [Actions]
<p>Hi everyone,</p>
<p>I used a Github action to build &amp; push an image to github container registry.<br>
But when I try to pull the image (either locally, or from a kubernetes cluster), I’m getting an error on the pulling of one layer <code>received unexpected HTTP status: 503 Response object too large</code><br>
So the status of this layers stay at “downloading”. Docker tries several time to download, but everytime, the same error.</p>
<p>All the other layers are properly pulled.<br>
The layer that cannot be pulled is pretty big (several Go).<br>
Is there a size limit by layer ? How can I get rid of this and manage to pull ?<br>
Is it the proper place to report this or is there another place to report bug/weird behaviour ? :slight_smile:</p>
<p>Here is the workflow I used to build &amp; push the image :</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/Hartorn/satisfactory-server/actions/runs/773649792/workflow" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/4953724/181098509-db7ffb4f-41de-4409-90c0-bd79a34a44a4.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/5/15e35513328b20196f4f3de569cf6f82829eca46_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/5/15e35513328b20196f4f3de569cf6f82829eca46_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/5/15e35513328b20196f4f3de569cf6f82829eca46.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/5/15e35513328b20196f4f3de569cf6f82829eca46_2_10x10.png"></div>

<h3><a href="https://github.com/Hartorn/satisfactory-server/actions/runs/773649792/workflow" target="_blank" rel="noopener nofollow ugc">Hartorn/satisfactory-server</a></h3>

<p><a href="https://github.com/Hartorn/satisfactory-server/actions/runs/773649792/workflow" target="_blank" rel="noopener nofollow ugc">ns/runs/773649792/workflow</a></p>

  <p><span class="label1">Contribute to Hartorn/satisfactory-server development by creating an account on GitHub.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I’m available to provide more details.</p>
<p>Cheers,<br>
Hartorn</p>

**Comment by jcansdale** [comment]
<p>Hi Hartorn,</p>
<aside class="quote no-group" data-username="Hartorn" data-post="1" data-topic="177017">
<div class="title">
<div class="quote-controls"></div>
 Hartorn:</div>
<blockquote>
<p>Is there a size limit by layer ? How can I get rid of this and manage to pull ?<br>
Is it the proper place to report this or is there another place to report bug/weird behaviour ? :slight_smile:</p>
</blockquote>
</aside>
<p>Due to the way our caching works, there is currently a limit of 4GB per layer. We’re hoping to fix this, but I don’t have an ETA for exactly when.</p>
<p>The workaround would be to break your image into layers of 4GB or less. Might this be feasible?</p>

**Comment by Hartorn** [comment]
<p>Hi <a class="mention" href="/jcansdale">@jcansdale</a>,</p>
<p>No not really. The layer is about 16go (yeah, quite big ^^’)<br>
I used a multi stage build to download and build binaries, and so the secret and auth are done into the first stage.</p>
<p>Nvm, not a big issue for me, I’m using a GCP container registry for that time, will do as a workaround.</p>
<p>I feel like it would be better to break the push command still atm, to avoid people being able to push, but not pull. It feels quite weird, and I took some time to say : “oh must be github”.</p>
<p>I have no strict timing on ETA, but I’m available to test if needed</p>
<p>Cheers.</p>

**Comment by markphelps** [comment, marked as answer]
<p><a class="mention" href="/hartorn">@Hartorn</a> this should be fixed now. Please give it a try again and let me know if its still a problem for you.</p>
<p>Thanks!</p>

**Comment by Hartorn** [comment]
<p><a class="mention" href="/markphelps">@markphelps</a> Pulling it right now.<br>
Seems a bit slow, but working nonetheless</p>
<p>Thanks !</p>
